tests/Swarrot/Processor/MaxExecutionTime/MaxExecutionTimeProcessorTest.php

```diff
@@ -54,4 +54,36 @@ public function test_count_default_messages_processed()
 
         $this->assertTrue(microtime(true) - $startTime > $maxExecutionTime);
     }
+
+    public function test_it_stops_after_slow_processing()
+    {
+        $message = new Message('body', [], 1);
+
+        $maxExecutionTime = 1;
+        $innerProcessor = $this->prophesize(ProcessorInterface::class);
+        $innerProcessor->process($message, ['max_execution_time' => $maxExecutionTime])->willReturn(true);
+
+        $logger = $this->prophesize(LoggerInterface::class);
+        $logger->info('[MaxExecutionTime] Max execution time has been reached', [
+            'max_execution_time' => $maxExecutionTime,
+            'swarrot_processor' => 'max_execution_time',
+        ])->shouldBeCalledTimes(1);
+
+        $processor = new MaxExecutionTimeProcessor($innerProcessor->reveal(), $logger->reveal());
+
+        $processor->initialize([]);
+
+        sleep(1);
+
+        $startTime = microtime(true);
+        $numberOfProcessedMessages = 0;
+        while (true) {
+            $numberOfProcessedMessages++;
+            if (false === $processor->process($message, ['max_execution_time' => $maxExecutionTime])) {
+                break;
+            }
+        }
+
+        $this->assertSame(1, $numberOfProcessedMessages);
+    }
```

Comment:
```suggestion
        $this->assertFalse($processor->process($message, ['max_execution_time' => $maxExecutionTime]))
```

I think this sould be enough?